vignettes/DBI.Rmd

```diff
@@ -13,109 +13,183 @@ vignette: >
 knitr::opts_chunk$set(echo = TRUE)
 ```
 
-The {DBI} package defines a common interface between the R and database management systems (DBMS).
-Hence the name: DBI stands for **d**ata**b**ase **i**nterface.
+### Who this tutorial is for
+- You have an understanding of R
+- You want to access or manipulate data in a database that may be on your machine or on a different computer on the internet
+- You have found libraries that use a higher level of abstraction, like [dplyer](missinglink), are not suitable for your purpose
+- You have an understanding of SQL, though it is not strictly necessary for basic data access
 
-Using DBI, developers can focus on the functionalities of their code, instead of setting up the infrastructure depending on the underlying database.
-This DBMS-agnostic approach is possible, because DBI works best with several other packages that act as drivers to absorb the peculiarities of the specific DBMSs.
+The {DBI} (**D**ata**B**ase **I**nterface) package provides a simple, consistent interface between R and database management systems (DBMS).  
+Each supported DBMS is supported by its own R package that implements the [DBI specification](https://cran.r-project.org/web/packages/DBI/vignettes/spec.html). You may need to install the package specific to your DBMS.
 
-These packages import {DBI} and implement its methods depending on the specific database management system.
-
-Currently, DBI works with the [many different database management systems](https://github.com/r-dbi/DBI/issues/274), e.g.:
+DBI currently supports 30 DBMS, including:
 
 * MySQL, using the R-package [RMySQL](https://github.com/r-dbi/RMySQL)
 * MariaDB, using the R-package [RMariaDB](https://github.com/r-dbi/RMariaDB)
-* Postgres, using and the R-package [RPostgres](https://github.com/r-dbi/RPostgres)
-* SQLite, using and the R-package [RSQLite](https://github.com/r-dbi/RSQLite)
+* Postgres, using the R-package [RPostgres](https://github.com/r-dbi/RPostgres)
+* SQLite, using the R-package [RSQLite](https://github.com/r-dbi/RSQLite)
 
-DBI offers a set of classes and methods that define what operations are possible and how they are performed:
+For the full list of supported DBMS visit [https://github.com/r-dbi/backends](https://github.com/r-dbi/backends)
 
-* connect/disconnect to the DBMS
-* create and execute statements in the DBMS
-* extract results/output from statements
-* error/exception handling
-* information (meta-data) from database objects
-* transaction management (optional)
+The functionality currently supported for each of these DBMS's includes:
```

Comment:
What's the Absolute Minimum that Every R User Who Works With a Database Must Know?

- Connect + disconnect, of course
- List tables?`Maybe
- Read a table
- Run a very basic query with `dbGetQuery()`

Everything else should be covered in part 2 of the tutorial:

- table details
- distinction between query (=returns data) and execution (=for side effect)
- parameterized queries
- `dbSendQuery()`/`dbSendStatement()` + `dbFetch()` + `dbClearResult()`, repeated `dbFetch()`
- transactions

Can we give a full overview here, explain the scope and link to the second part?